notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.notifier.modules
+
+import com.atlassian.jira.rest.client.api.JiraRestClient
+import com.atlassian.jira.rest.client.api.RestClientException
+import com.atlassian.jira.rest.client.api.domain.BasicIssue
+import com.atlassian.jira.rest.client.api.domain.Comment
+import com.atlassian.jira.rest.client.api.domain.IssueType
+import com.atlassian.jira.rest.client.api.domain.input.IssueInputBuilder
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory
+
+import java.net.URI
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.config.JiraConfiguration
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+
+class JiraNotifier(
+    val config: JiraConfiguration,
+    private val restClient: JiraRestClient = AsynchronousJiraRestClientFactory()
+        .createWithBasicHttpAuthentication(URI(config.host), config.username, config.password)
+) {
+    /** Concurrent map of all available issue types for each project */
+    private val projectIssueTypes: ConcurrentMap<String, Map<String, IssueType>> = ConcurrentHashMap()
```

Comment:
Could you elaborate why this needs to be a concurrent map? I think, the check whether a project's issue types have already been loaded is not actually thread-safe, since it is not done in an atomic way.

Reply:
Honestly, there is no reason to use the concurrent map here. I just changed it to a `MutableMap`.

Reply:
While at it, please omit the explicit type by rewriting this as `private val projectIssueTypes = mutableMapOf<String, Map<String, IssueType>>()`.